library/std/src/collections/hash/set.rs

```diff
@@ -809,11 +809,11 @@ where
         other.is_subset(self)
     }
 
-    /// Adds a value to the set.
+    /// Inserts a value to the set.
     ///
-    /// If the set did not have this value present, `true` is returned.
+    /// If the insertion changed the set, `true` is returned.
```

Comment:
i prefer still specifying the state before the insertion. it also requires some thinking to decode what is meant by "changed".

```suggestion
    /// If the the value was absent, i.e. insertion changed the set, `true` is returned.
```

Reply:
What I care most about is the meaning of the returned value.
If there are descriptions for both cases, they should at least not be redundant.

I want a reason, why each value is returned in each case.

Maybe also "Returns if insertion was successful".
This seems like a very straightforward description. But maybe it's not correct when using the mathematical definition of a set.
You can insert the same value into a set multiple times, it wouldn't change, and would still be considered as successful, I guess.

Reply:
@porky11 The current wording in this PR maybe clear to *you* but it is not clear to *me*, and documentation should be clear to *all of us*. Documentation is not the same as coding and redundancy for clarity is not a bad thing.

By only saying "insertion changed the set", *I* need to do the calculation of "What is meant by changed? The final state is value has been inserted, and a hash-set can either contain a value or not, so 'changed' must mean the initial state is the value was absent, OK!". That's a totally unnecessary "make me think" exercise that can be avoided by just including the "value was absent" in the doc and makes everyone happy.